hollow/src/test/java/com/netflix/hollow/api/codegen/AbstractHollowAPIGeneratorTest.java

```diff
@@ -55,7 +61,15 @@ void assertNonEmptyFileExists(String relativePath) throws IOException {
         assertTrue("File at " + relativePath + " should exist", f.exists() && f.length() > 0L);
     }
 
-    void assertFileDoesNotExist(String relativePath) throws IOException {
+    void assertClassHasHollowTypeName(String clazz, String typeName) throws IOException, ClassNotFoundException {
```

Comment:
Couldn't find a good way to test what we really want (that we can create generated indices) so went with the best I could - verify that a HollowTypeName is set.

Reply:
That seems reasonable.  We can work on an additional fix to the indexes.  If you look in the test class `com.netflix.hollow.api.consumer.index.DataModel.Consumer` we can add something explicit.

Reply:
I don't think I understand your comment - is there a way to test the generated index, or are you suggesting adding a test to make sure that the HollowTypeName is picked up by indexes?

Reply:
I found the cause.  Select path never quite worked correctly, it currently requires the path have a last segment reference a field value, and the ordinal returned will correspond to the enclosing record.

The problematic line is at https://github.com/Netflix/hollow/blob/master/hollow/src/main/java/com/netflix/hollow/core/index/traversal/TraversalTreeBuilder.java#L103:
```java
                            if(schema.getFieldType(fieldSchemaPositions[i]) == FieldType.REFERENCE) {
                                fieldSchemaPositions[i] = -1;
                                fieldTypeDataAccess[i] = getChildDataAccess(typeDataAccess, pathElement);
                            } else {
                                fieldSchemaPositions[i] = schema.getPosition(pathElement);
                                fieldTypeDataAccess[i] = typeDataAccess;
                            }
```

It should be:
```java
                            if(schema.getFieldType(pathElement) == FieldType.REFERENCE) {
```